packages/core/src/actions.ts

```diff
@@ -467,7 +468,9 @@ export function resolveStop<TContext, TEvent extends EventObject>(
  * @param assignment An object that represents the partial context to update.
  */
 export const assign = <TContext, TEvent extends EventObject = EventObject>(
-  assignment: Assigner<TContext, TEvent> | PropertyAssigner<TContext, TEvent>
+  assignment:
+    | Assigner<LowInfer<TContext>, TEvent>
+    | PropertyAssigner<LowInfer<TContext>, TEvent>
```

Comment:
This has some drawbacks when it comes to inferring standalone `assign`s, like here:
```ts
const a1 = assign(/*...*/)
```

When `Assigner` is used then it works just fine:
```ts
const a1 = assign((ctx: { counter: number }) => ({})) // TContext inferred to `{ counter: number }`
```
but when it comes to `PropertyAssigner` then it doesn't infer `TContext` and it leaves it as `unknown` (the internal partial assigned function stays types correctly, it's about what the `typeof a1` is):
```ts
const a1 = assign({
  counter: (context: { counter: number }) => context.counter + 1,
})
```
It also doesn't work, in this context, with static values:
```ts
const a1 = assign({
  counter: 100,
})
```

I think that `LowInfer` combined with reverse mapped types is broken in TS, a minimal repro case of this can be found here: [TS playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAMg9gdwJIDsBmEBOAeAKgPigF4pcoAyKAbwF8BuAKAdEigCUIBbAQzD0JJUGUKAG0A0lACWKKAGsIIOGlIBdAFykJqxvSYATCAGMANt0zQjcFAGdgUIxe7AIAeQBGAK038AFHC9NDh4+eGR0LH58AEpiQlxGBitbewsbYgcnFw9PXyoHOABXFBdMTQAGABooblsELE0AIjQ4OEaoGmjGAHpukREAPQB+BiA)